### PR TITLE
PHP-591 => master

### PR DESCRIPTION
**Pull request by sidibos**
Modified Cloud Formation Template to adopt on the new API Gateway stack

### CODE REVIEW

File: cloudformation-template.template [hunk 1]
```diff
@@ -1,17 +1,33 @@
 {
   "AWSTemplateFormatVersion": "2010-09-09",
-  "Parameters": {
-    "s3bucketName": {
+  "Parameters" : {
+    "OriginAPIToken" : {
+      "Type" : "String",
+      "Default" : "12345678",
```
Comment (ookejimift) [inline cloudformation-template.template]: do we want to check-in a key into git, I know it is meant to be changed but users may not change it. should it be a mandatory field with no default value instead

File: cloudformation-template.template [hunk 2]
```diff
@@ -22,6 +38,52 @@
...
+    "TopAnyMethod": {
+      "Type": "AWS::ApiGateway::Method",
+      "Properties": {
+        "RestApiId": {
+          "Ref": "LanternApiGateway"
+        },
+        "ResourceId": { "Fn::GetAtt": ["LanternApiGateway", "RootResourceId"] },
+        "HttpMethod": "ANY",
+        "RequestParameters": {
+          "method.request.header.x-api-key": true
```
Comment (ookejimift) [inline cloudformation-template.template]: FYI, this bit will break existing clients when it goes live - we may need to coordinate and apply the new header to lantern and lantern-dashboard on go live of this

**Comment by sidibos** [comment]
@ookejimift we don't have to commit a key to git - I will make this field mandatory instead

**Comment by sidibos** [comment]
@ookejimift I do agree we will have to coordinate before this goes live, so the corresponding changes can be made on the client side, the API KEY, the new header value 'x-api-key'.

**Pull request merged**